packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js

```diff
@@ -38,10 +38,15 @@ export default function usePatternDetails( postType, postId ) {
 	const { currentTheme, patternCategories } = useSelect( ( select ) => {
 		const { getCurrentTheme, getUserPatternCategories } =
 			select( coreStore );
+		const userPatternCategories = getUserPatternCategories();
+		const categories = new Map();
+		userPatternCategories.forEach( ( userCategory ) =>
+			categories.set( userCategory.id, userCategory )
+		);
```

Comment:
This will still return a new value every time `mapSelect` is called. You should see the same `useSelect` [notice](https://github.com/WordPress/gutenberg/pull/53666) where this hook is used.

We should generally avoid mapping/grouping the items inside the selectors to avoid bugs like #53596.

P.S. The `usePatterns` already had this issue, and I've left a comment on the [original PR](https://github.com/WordPress/gutenberg/pull/52303#issuecomment-1678885719). Cc @kevin940726.

Reply:
Appreciate you following up on the useSelect notices here @Mamaduka 👍 

I was hoping to have a look into these this afternoon.

Fixing up `use-pattern-details.js` is pretty straightforward, it might take me a bit longer to untangle the `usePatterns` hook though.

Reply:
@aaronrobertshaw, yes, the `usePatterns` will take more refactoring to untangle :(

Reply:
I'll throw up a PR for `usePatternDetails` shortly so that small chunk isn't held up by the larger refactor.

Reply:
Thanks @Mamaduka! I knew that there was a ping from you but I couldn't remember where 😅.

Yeah, the `use-patterns` one was my bad, but I don't think it's the bottleneck in most cases. Re-renders aren't bad IMO, only when the computation is heavy.

However, I agree we can optimize it, usually done by memoizing intermediate results. I'm looking into ways to memoize the selected data using `rememo` and let's see if it helps!

Reply:
@kevin940726 is already working on `usePatterns` so I'll aim to help him on that front.

Reply:
Thank you both! Happy to help out with testing/reviews.